crates/aptos-crypto/src/webauthn/webauthn_p256_keys.rs

```diff
@@ -0,0 +1,286 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! This file implements traits for WebAuthn P256 private keys and public keys.
+
+#[cfg(any(test, feature = "fuzzing"))]
+use crate::test_utils::{self, KeyPair};
+use crate::{
+    hash::CryptoHash,
+    p256_ecdsa::{P256PrivateKey, P256PublicKey, P256_PUBLIC_KEY_LENGTH},
+    traits::*,
+    webauthn::webauthn_p256_sigs::WebAuthnP256Signature,
+};
+use aptos_crypto_derive::{DeserializeKey, SerializeKey, SilentDebug, SilentDisplay};
+use core::convert::TryFrom;
+use p256::{self, ecdsa};
+#[cfg(any(test, feature = "fuzzing"))]
+use proptest::prelude::*;
+use serde::Serialize;
+use std::fmt;
+
+
+/// A WebAuthn P256 private key
+#[derive(DeserializeKey, SerializeKey, SilentDebug, SilentDisplay)]
+pub struct WebAuthnP256PrivateKey(pub(crate) P256PrivateKey);
+
+impl private::Sealed for WebAuthnP256PrivateKey {}
+
+#[cfg(feature = "assert-private-keys-not-cloneable")]
+static_assertions::assert_not_impl_any!(WebAuthnP256PrivateKey: Clone);
+
+#[cfg(any(test, feature = "cloneable-private-keys"))]
+impl Clone for WebAuthnP256PrivateKey {
+    fn clone(&self) -> Self {
+        let serialized: &[u8] = &(self.to_bytes());
+        WebAuthnP256PrivateKey::try_from(serialized).unwrap()
+    }
+}
+
+/// A WebAuthn P256 public key
+#[derive(DeserializeKey, Clone, SerializeKey)]
+pub struct WebAuthnP256PublicKey(pub(crate) P256PublicKey);
+
+impl private::Sealed for WebAuthnP256PublicKey {}
+
+impl WebAuthnP256PrivateKey {
```

Comment:
I think you actually want a From< P256PrivateKey> for this struct instead of copy the code here.